src/metabase/events/notification.clj

```diff
@@ -52,23 +56,68 @@
   (when (supported-topics topic)
     (models.notification/notifications-for-event topic)))
 
+(defn- app-name-trs
+  "Return the user configured application name, or Metabase translated
+  via trs if a name isn't configured."
+  []
+  (or (public-settings/application-name)
+      (trs "Metabase")))
+
+(defn- site-name
+  []
+  (or (public-settings/site-name) (trs "Unknown")))
+
+(defn- join-url
+  [new-user]
+  (let [reset-token               (user/set-password-reset-token! (:id new-user))
+        should-link-to-login-page (and (public-settings/sso-enabled?)
+                                       (not (public-settings/enable-password-login)))]
+    (if should-link-to-login-page
+      (str (public-settings/site-url) "/auth/login")
+      ;; NOTE: the new user join url is just a password reset with an indicator that this is a first time user
+      (str (user/form-password-reset-url reset-token) "#new"))))
+
+(defn- extra-context
+  "Returns a map of extra context for a given topic and event-info.
+  Extra are set of contexts that are specific to a certain emails.
+  Currently we need it to support usecases that our template engines doesn't support such as i18n,
+  but ideally this should be part of the template."
+  [topic event-info]
+  (case topic
+    :event/user-invited
+    {:user-invited-today         (t/format "MMM'&nbsp;'dd,'&nbsp;'yyyy" (t/zoned-date-time))
+     :user-invited-email-subject (trs "You''re invited to join {0}''s {1}" (site-name) (app-name-trs))
+     ;; TODO test that this link works for real
+     ;; TODO this should link to login page if sso is eanbled
+     :user-invited-join-url      (join-url (:object event-info))}
+    {}))
+
 (defn- enriched-event-info
   [topic event-info]
   ;; DO NOT delete or rename these fields, they are used in the notification templates
-  {:settings    {:application-name (public-settings/application-name)
-                 :site-name        (public-settings/site-name)}
+  {:context     {:application-name (public-settings/application-name)
+                 :site-name        (site-name)
+                 :extra            (extra-context topic event-info)}
    :event-info  (cond->> event-info
                   (some? (events.schema/topic->schema topic))
                   (hydrate! (events.schema/topic->schema topic)))
    :event-topic topic})
 
+(def ^:dynamic *skip-sending-notification?*
+  "Used as a hack for when we need to skip sending notifications for certain events.
+
+  It's an escape hatch until we implement conditional notifications."
```

Comment:
Ah, if it's temporary I guess its OK as long as we're testing this behavior in a way that's robust to any async changes.